contrib/src/main/java/com/datatorrent/contrib/enrich/DelimitedFSLoader.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.enrich;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.solr.internal.csv.CSVParser;
+import org.apache.solr.internal.csv.CSVStrategy;
+
+import com.google.common.collect.Maps;
+
+import com.datatorrent.contrib.parser.AbstractCsvParser.FIELD_TYPE;
+import com.datatorrent.contrib.parser.AbstractCsvParser.Field;
+
+/**
+ * This implementation of {@link FSLoader} is used to load data from delimited
+ * file.
+ *
+ */
+public class DelimitedFSLoader extends FSLoader
+{
+
+  /**
+   * CSVStrategy Represents the strategy for a CSV
+   */
+  private transient CSVStrategy strategy;
+  /**
+   * List of Fields. Each Field has a name and a type.
+   */
+  private transient List<Field> fields;
+  /**
+   * Indicates whether first line of the file is a header. Default is false
+   */
+  private transient boolean hasHeader;
+
+  /**
+   * Array containing headers
+   */
+  private transient String[] header;
+
+  private static final Logger logger = LoggerFactory.getLogger(DelimitedFSLoader.class);
+
+  /**
+   * 
+   * @param fieldDescription
+   *          String representing fields and its type in the same order as it
+   *          appears in the records. E.g.
+   *          empId:INTEGER,address:STRING,phone:LONG Supported data types are
+   *          BOOLEAN, INTEGER, FLOAT, DOUBLE, LONG, SHORT, CHARACTER, STRING
+   */
+  public DelimitedFSLoader(String fieldDescription)
```

Comment:
You need to have empty constructor for this and let user set the properties for DelimitedFSLoader using properties file.
Have setter method for individual parameters for this FSLoader.